--- lttng-tools-2.1.1/configure.ac.orig	2013-02-04 10:06:56.183426571 +0000
+++ lttng-tools-2.1.1/configure.ac	2013-02-04 10:07:20.863921050 +0000
@@ -249,7 +249,7 @@
 
 CFLAGS="-Wall $CFLAGS -g -fno-strict-aliasing"
 
-DEFAULT_INCLUDES="-I\$(top_srcdir) -I\$(top_builddir) -I\$(top_builddir)/src -I\$(top_builddir)/include -include config.h"
+DEFAULT_INCLUDES="-I\$(top_srcdir) -I\$(top_srcdir)/include -I\$(top_builddir) -I\$(top_builddir)/src -I\$(top_builddir)/include -include config.h"
 
 lttngincludedir="${includedir}/lttng"
 
