--- libffi-3.0.13/configure.ac.orig	2013-12-19 11:29:01.658038148 +0100
+++ libffi-3.0.13/configure.ac	2013-12-19 11:29:49.070036747 +0100
@@ -546,11 +546,11 @@
     toolexecdir='$(libdir)/gcc-lib/$(target_alias)'
     toolexeclibdir='$(libdir)'
   fi
-  multi_os_directory=`$CC -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
+  #multi_os_directory=`$CC -print-multi-os-directory`
+  #case $multi_os_directory in
+  #  .) ;; # Avoid trailing /.
+  #  ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
+  #esac
   AC_SUBST(toolexecdir)
 else
   toolexeclibdir='$(libdir)'
--- libffi-3.0.13/configure.orig	2013-12-19 11:28:50.710038471 +0100
+++ libffi-3.0.13/configure	2013-12-19 11:29:36.738037111 +0100
@@ -14840,11 +14840,11 @@
     toolexecdir='$(libdir)/gcc-lib/$(target_alias)'
     toolexeclibdir='$(libdir)'
   fi
-  multi_os_directory=`$CC -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
+  #multi_os_directory=`$CC -print-multi-os-directory`
+  #case $multi_os_directory in
+  #  .) ;; # Avoid trailing /.
+  #  ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
+  #esac
 
 else
   toolexeclibdir='$(libdir)'
