--- lttng-tools-2.0-pre14/ltt-sessiond/main.c~orig	2011-11-09 15:07:23.819627443 +0100
+++ lttng-tools-2.0-pre14/ltt-sessiond/main.c	2011-11-09 15:08:59.975622813 +0100
@@ -193,16 +193,16 @@
 
 	for (i = ARRAY_SIZE(kernel_modules_list) - 1; i >= 0; i--) {
 		ret = snprintf(modprobe, sizeof(modprobe),
-				"/sbin/modprobe --remove --quiet %s",
+				"/sbin/modprobe -r -q %s",
 				kernel_modules_list[i].name);
 		if (ret < 0) {
-			perror("snprintf modprobe --remove");
+			perror("snprintf modprobe -r");
 			goto error;
 		}
 		modprobe[sizeof(modprobe) - 1] = '\0';
 		ret = system(modprobe);
 		if (ret == -1) {
-			ERR("Unable to launch modprobe --remove for module %s",
+			ERR("Unable to launch modprobe -r for module %s",
 					kernel_modules_list[i].name);
 		} else if (kernel_modules_list[i].required
 				&& WEXITSTATUS(ret) != 0) {
@@ -1322,9 +1322,9 @@
 		 * Exec kconsumerd.
 		 */
 		if (opt_verbose > 1 || opt_verbose_kconsumerd) {
-			verbosity = "--verbose";
+			verbosity = "-v";
 		} else {
-			verbosity = "--quiet";
+			verbosity = "-q";
 		}
 		execl(INSTALL_BIN_PATH "/ltt-kconsumerd",
 				"ltt-kconsumerd", verbosity, NULL);
@@ -1397,7 +1397,7 @@
 	for (i = 0; i < ARRAY_SIZE(kernel_modules_list); i++) {
 		ret = snprintf(modprobe, sizeof(modprobe),
 			"/sbin/modprobe %s%s",
-			kernel_modules_list[i].required ? "" : "--quiet ",
+			kernel_modules_list[i].required ? "" : "-q ",
 			kernel_modules_list[i].name);
 		if (ret < 0) {
 			perror("snprintf modprobe");
