--- glib-2.34.0/tests/threadpool-test.c.orig	2012-10-10 13:13:53.705427296 +0200
+++ glib-2.34.0/tests/threadpool-test.c	2012-10-10 13:13:53.705427296 +0200
@@ -5,7 +5,7 @@
 
 #include <glib.h>
 
-/* #define DEBUG 1 */
+#define DEBUG 1
 
 #ifdef DEBUG
 # define DEBUG_MSG(args) g_printerr args ; g_printerr ("\n");
@@ -113,10 +113,10 @@
        if (g_thread_pool_get_num_unused_threads () == 0)
          break;
 
-       DEBUG_MSG (("[unused] waiting ONE second for threads to die"));
+       DEBUG_MSG (("[unused] waiting FIVE seconds for threads to die"));
 
        /* Some time for threads to die. */
-       g_usleep (G_USEC_PER_SEC);
+       g_usleep (5 * G_USEC_PER_SEC);
      }
 
    DEBUG_MSG (("[unused] stopped idle threads, %d remain",
