--- netperf-2.4.5/configure.orig	2010-06-15 13:36:14.000000000 +0200
+++ netperf-2.4.5/configure	2010-06-15 13:36:25.000000000 +0200
@@ -1660,7 +1660,7 @@
 # Let the site file select an alternate cache file if it wants to.
 # Prefer explicitly selected file to automatically selected ones.
 if test -n "$CONFIG_SITE"; then
-  set x "$CONFIG_SITE"
+  set x $CONFIG_SITE
 elif test "x$prefix" != xNONE; then
   set x "$prefix/share/config.site" "$prefix/etc/config.site"
 else
