diff -urN GConf-3.2.0.orig/backends/Makefile.am GConf-3.2.0/backends/Makefile.am
--- GConf-3.2.0.orig/backends/Makefile.am	2011-09-29 09:05:01.514883595 +0200
+++ GConf-3.2.0/backends/Makefile.am	2011-09-29 09:05:38.656171231 +0200
@@ -1,4 +1,4 @@
-INCLUDES= -I$(top_srcdir) -I$(top_builddir) -I$(top_builddir)/gconf \
+AM_CPPFLAGS= -I$(top_srcdir) -I$(top_builddir) -I$(top_builddir)/gconf \
 	$(DEPENDENT_WITH_XML_CFLAGS) \
 	$(DEPENDENT_ORBIT_CFLAGS) \
 	-DGCONF_ENABLE_INTERNALS=1 -DG_LOG_DOMAIN=\"GConf-Backends\"
diff -urN GConf-3.2.0.orig/defaults/Makefile.am GConf-3.2.0/defaults/Makefile.am
--- GConf-3.2.0.orig/defaults/Makefile.am	2011-09-29 09:05:01.524883934 +0200
+++ GConf-3.2.0/defaults/Makefile.am	2011-09-29 09:05:56.566792197 +0200
@@ -12,7 +12,7 @@
 	gconf-defaults-glue.h \
 	gconf-defaults-main.c
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir) \
 	-I$(top_builddir) \
 	-DSYSGCONFDIR=\"$(sysgconfdir)\" \
diff -urN GConf-3.2.0.orig/examples/Makefile.am GConf-3.2.0/examples/Makefile.am
--- GConf-3.2.0.orig/examples/Makefile.am	2011-09-29 09:05:01.524883934 +0200
+++ GConf-3.2.0/examples/Makefile.am	2011-09-29 09:06:25.797805671 +0200
@@ -1,4 +1,4 @@
-INCLUDES=-I$(top_srcdir) -I$(top_builddir) \
+AM_CPPFLAGS=-I$(top_srcdir) -I$(top_builddir) \
 	$(DEPENDENT_WITH_GTK_CFLAGS) \
         -DGCONF_SRCDIR=\""$(absolute_top_srcdir)"\" -DGCONF_SYSCONFDIR=\""$(sysgconfdir)"\"
 
diff -urN GConf-3.2.0.orig/Makefile.am GConf-3.2.0/Makefile.am
--- GConf-3.2.0.orig/Makefile.am	2011-09-29 09:05:01.524883934 +0200
+++ GConf-3.2.0/Makefile.am	2011-09-29 09:05:16.585406054 +0200
@@ -54,3 +54,4 @@
 		echo A git clone is required to generate a ChangeLog >&2; \
 	fi
 
+ACLOCAL_AMFLAGS = -I m4

diff -urN GConf-3.2.0.orig/gconf/Makefile.am GConf-3.2.0/gconf/Makefile.am
--- GConf-3.2.0.orig/gconf/Makefile.am	2011-09-29 09:13:51.513266147 +0200
+++ GConf-3.2.0/gconf/Makefile.am	2011-09-29 09:14:13.544030634 +0200
@@ -2,7 +2,7 @@
 
 GCONFD_BINARY_NAME=gconfd-2
 
-INCLUDES=								\
+AM_CPPFLAGS=								\
 	-I$(top_srcdir)							\
 	-I$(top_builddir)						\
 	$(DEPENDENT_WITH_XML_AND_GTK_CFLAGS) 				\

diff -urN GConf-3.2.0.orig/tests/Makefile.am GConf-3.2.0/tests/Makefile.am
--- GConf-3.2.0.orig/tests/Makefile.am	2011-09-29 09:13:51.523266498 +0200
+++ GConf-3.2.0/tests/Makefile.am	2011-09-29 09:14:24.004393631 +0200
@@ -1,7 +1,7 @@
 
 EFENCE=
 
-INCLUDES = -I$(top_srcdir) -I$(top_builddir) \
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_builddir) \
 	 $(DEPENDENT_CFLAGS) \
 	 -DG_LOG_DOMAIN=\"GConf-Tests\" -DGCONF_ENABLE_INTERNALS=1
 

