--- lftp-4.6.0/src/ftpclass.cc.orig	2015-01-08 11:52:08.701226756 +0100
+++ lftp-4.6.0/src/ftpclass.cc	2015-01-08 11:52:32.721226408 +0100
@@ -1610,12 +1610,12 @@
 
       if(expect->Has(Expect::FEAT)
       || expect->Has(Expect::OPTS_UTF8)
-      || expect->Has(Expect::LANG)
-      || expect->Has(Expect::PROT))
+      || expect->Has(Expect::LANG))
 	 goto usual_return;
 
 #if USE_SSL
-      if(expect->Has(Expect::CCC))
+      if(expect->Has(Expect::CCC)
+      || expect->Has(Expect::PROT))
 	 goto usual_return;
 #endif // USE_SSL
 
