--- gobject-introspection-0.9.10/gir/Makefile.am.orig	2011-09-29 10:33:44.730060277 +0200
+++ gobject-introspection-0.9.10/gir/Makefile.am	2011-09-29 10:34:09.670929828 +0200
@@ -38,7 +38,7 @@
 
 # glib
 GLIB_INCLUDEDIR=$(shell pkg-config --variable=includedir glib-2.0)/glib-2.0
-GLIB_LIBDIR=$(shell pkg-config --variable=libdir glib-2.0)
+GLIB_LIBDIR=$(STAGE_DIR)/native/lib
 
 if OS_WIN32
 GLIB_LIBRARY=libglib-2.0-0
