diff -urN libpng-1.5.14~orig/Makefile.am libpng-1.5.14/Makefile.am
--- libpng-1.5.14~orig/Makefile.am	2012-12-27 16:58:18.000000000 +0100
+++ libpng-1.5.14/Makefile.am	2014-01-24 11:57:31.147493827 +0100
@@ -50,7 +50,8 @@
 libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_CPPFLAGS = @LIBPNG_DEFINES@
 
 libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_LDFLAGS = -no-undefined -export-dynamic \
-	-version-number @PNGLIB_MAJOR@@PNGLIB_MINOR@:@PNGLIB_RELEASE@:0
+	-version-number @PNGLIB_MAJOR@@PNGLIB_MINOR@:@PNGLIB_RELEASE@:0 \
+	-bindir "$(bindir)"
 
 if HAVE_LD_VERSION_SCRIPT
 #   Versioned symbols and restricted exports
diff -urN libpng-1.5.14~orig/Makefile.in libpng-1.5.14/Makefile.in
--- libpng-1.5.14~orig/Makefile.in	2013-01-24 03:59:12.000000000 +0100
+++ libpng-1.5.14/Makefile.in	2014-01-24 11:57:42.159493599 +0100
@@ -395,6 +395,7 @@
 libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_LDFLAGS = -no-undefined \
 	-export-dynamic -version-number \
 	@PNGLIB_MAJOR@@PNGLIB_MINOR@:@PNGLIB_RELEASE@:0 \
+	-bindir "$(bindir)" \
 	$(am__append_2) $(am__append_3) $(am__append_4)
 @HAVE_LD_VERSION_SCRIPT_FALSE@libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_DEPENDENCIES = libpng.sym
 @HAVE_LD_VERSION_SCRIPT_TRUE@libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_DEPENDENCIES = libpng.vers
