--- valgrind-3.8.1/configure.in~orig	2012-11-09 11:02:58.179810766 +0100
+++ valgrind-3.8.1/configure.in	2012-11-09 11:04:01.203809185 +0100
@@ -191,6 +191,19 @@
 	ARCH_MAX="arm"
 	;;
 
+     arm)
+	case "${host_vendor}" in
+	     cortexa*)
+		AC_MSG_RESULT([ok (${host_cpu}-${host_vendor})])
+		ARCH_MAX="arm"
+	        ;;
+	     *) 
+		AC_MSG_RESULT([no (${host_cpu}-${host_vendor})])
+		AC_MSG_ERROR([Unsupported host architecture. Sorry])
+		;;
+	esac
+	;;
+
      mips)
 	AC_MSG_RESULT([ok (${host_cpu})])
 	ARCH_MAX="mips32"
