--- libffi-3.0.12/configure.ac~orig	2013-04-29 19:57:18.016364692 +0200
+++ libffi-3.0.12/configure.ac	2013-04-29 19:57:29.679911231 +0200
@@ -541,11 +541,11 @@
     toolexecdir='$(libdir)/gcc-lib/$(target_alias)'
     toolexeclibdir='$(libdir)'
   fi
-  multi_os_directory=`$CC -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    *) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
+  #multi_os_directory=`$CC -print-multi-os-directory`
+  #case $multi_os_directory in
+  #  .) ;; # Avoid trailing /.
+  #  *) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
+  #esac
   AC_SUBST(toolexecdir)
 else
   toolexeclibdir='$(libdir)'
--- libffi-3.0.12/configure~	2013-04-29 19:59:42.274780036 +0200
+++ libffi-3.0.12/configure	2013-04-29 20:10:49.097021366 +0200
@@ -14826,11 +14826,11 @@
     toolexecdir='$(libdir)/gcc-lib/$(target_alias)'
     toolexeclibdir='$(libdir)'
   fi
-  multi_os_directory=`$CC -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    *) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
+  #multi_os_directory=`$CC -print-multi-os-directory`
+  #case $multi_os_directory in
+  #  .) ;; # Avoid trailing /.
+  #  *) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
+  #esac
 
 else
   toolexeclibdir='$(libdir)'
