--- dhcp-3.1-ESV-R3~orig/configure	2013-10-14 10:02:43.000000000 +0200
+++ dhcp-3.1-ESV-R3/configure	2013-10-14 10:10:26.161601346 +0200
@@ -1,7 +1,5 @@
 #!/bin/sh
 
-#sysname=$1
-
 while [ $# != 0 ]; do
   if [ x$1 = x--with-nsupdate ]; then
     echo "nsupdate is always built now."
@@ -20,6 +18,16 @@
     shift
   elif [ x$sysname = x ]; then
     sysname=$1
+    if [ x$sysname = xlinux-2.0 ] ; then
+      major=2 ; minor=0
+    elif [ x$sysname = xlinux-2.1 ] ; then
+      major=2 ; minor=1
+    elif [ x$sysname = xlinux-2.2 ] ; then
+      major=2 ; minor=2
+    elif [ x$sysname = xlinux ] ; then
+      sysname=linux-2.2
+      major=${2-3} ; minor=${3-0}
+    fi
   else
     echo "Unexpected argument: $1"
   fi
