--- dbus-1.4.16/configure.orig	2011-10-03 14:22:03.805260050 +0200
+++ dbus-1.4.16/configure	2011-10-03 14:25:18.074153601 +0200
@@ -22088,15 +22088,7 @@
 
 
 #### Find socket directories
-if ! test -z "$TMPDIR" ; then
-   DEFAULT_SOCKET_DIR=$TMPDIR
-elif ! test -z "$TEMP" ; then
-   DEFAULT_SOCKET_DIR=$TEMP
-elif ! test -z "$TMP" ; then
-   DEFAULT_SOCKET_DIR=$TMP
-else
-   DEFAULT_SOCKET_DIR=/tmp
-fi
+DEFAULT_SOCKET_DIR=/tmp
 
 DEFAULT_SOCKET_DIR=`echo $DEFAULT_SOCKET_DIR | sed 's/+/%2B/g'`
 

