--- glib-2.39.2/glib/grand.c~orig	2014-04-15 08:37:47.810830637 +0200
+++ glib-2.39.2/glib/grand.c	2014-04-15 08:37:51.922830525 +0200
@@ -58,6 +58,7 @@
 
 #ifdef G_OS_WIN32
 #include <stdlib.h>
+#include <process.h>
 #endif
 
 /**
@@ -264,10 +265,12 @@
       seed[3] = getppid ();
     }
 #else /* G_OS_WIN32 */
-  gint i;
-
-  for (i = 0; i < G_N_ELEMENTS (seed); i++)
-    rand_s (&seed[i]);
+  GTimeVal now;
+  g_get_current_time (&now);
+  seed[0] = now.tv_sec;
+  seed[1] = now.tv_usec;
+  seed[2] = getpid ();
+  seed[3] = 0;
 #endif
 
   return g_rand_new_with_seed_array (seed, 4);
