--- util-linux-2.20/configure.ac~orig	2013-04-23 09:15:01.916785158 +0200
+++ util-linux-2.20/configure.ac	2013-04-23 09:17:54.548780825 +0200
@@ -62,17 +62,32 @@
   esac ;;
 esac
 
+bindirname=`basename "$bindir"`
+AC_SUBST([bindirname])
+
+sbindirname=`basename "$sbindir"`
+AC_SUBST([sbindirname])
+
 libdirname=`basename "$libdir"`
 AC_SUBST([libdirname])
 
 # The original default values of {bin,sbin,lib}dir
-usrbin_execdir='${exec_prefix}/bin'
+if test -z "${usrbin_execdir}" ; then
+  usrbin_execdir='${exec_prefix}/'$bindirname
+fi
 AC_SUBST([usrbin_execdir])
 
-usrsbin_execdir='${exec_prefix}/sbin'
+if test -z "${usrsbin_execdir}" ; then
+  usrsbin_execdir='${exec_prefix}/'$sbindirname
+fi
 AC_SUBST([usrsbin_execdir])
 
-usrlib_execdir='${exec_prefix}/'$libdirname
+case "$libdir" in
+  '${exec_prefix}'/* | '${prefix}'/* | /usr/*)
+    usrlib_execdir=$libdir ;;
+  *)
+    usrlib_execdir='${exec_prefix}/'$libdirname ;;
+esac
 AC_SUBST([usrlib_execdir])
 
 AM_PROG_CC_C_O
