--- mysql-5.6.13/extra/CMakeLists.txt.orig	2013-08-15 13:40:58.821224585 +0200
+++ mysql-5.6.13/extra/CMakeLists.txt	2013-08-15 13:41:26.689223761 +0200
@@ -26,16 +26,14 @@
 # Default install component for the files is Server here
 SET(MYSQL_INSTALL_COMPONENT Server)
 
-IF(NOT CMAKE_CROSSCOMPILING)
- ADD_EXECUTABLE(comp_err comp_err.c)
- TARGET_LINK_LIBRARIES(comp_err mysys mysys_ssl)
- SET_TARGET_PROPERTIES(comp_err PROPERTIES LINKER_LANGUAGE CXX)
-ENDIF()
+ADD_EXECUTABLE(comp_err comp_err.c)
+TARGET_LINK_LIBRARIES(comp_err mysys mysys_ssl)
+SET_TARGET_PROPERTIES(comp_err PROPERTIES LINKER_LANGUAGE CXX)
 
 
 ADD_CUSTOM_COMMAND(OUTPUT ${PROJECT_BINARY_DIR}/include/mysqld_error.h 
                    ${PROJECT_BINARY_DIR}/sql/share/english/errmsg.sys
-                   COMMAND comp_err
+                   COMMAND ./comp_err
                     --charset=${PROJECT_SOURCE_DIR}/sql/share/charsets 
                     --out-dir=${PROJECT_BINARY_DIR}/sql/share/
                     --header_file=${PROJECT_BINARY_DIR}/include/mysqld_error.h 
