Upstream-Status: Submitted http://mm.icann.org/pipermail/tz/2016-February/023278.html

Fixes needed to build for mingw32.

Signed-off-by: Esben Haabendal <eha@deif.com>

diff -urN tz-2016a~orig/Makefile tz-2016a/Makefile
--- tz-2016a~orig/Makefile	2016-01-27 08:30:16.000000000 +0100
+++ tz-2016a/Makefile	2016-02-23 10:48:52.017740672 +0100
@@ -125,6 +125,7 @@
 #  -DHAVE_SYS_WAIT_H=0 if your compiler lacks a "sys/wait.h"
 #  -DHAVE_TZSET=0 if your system lacks a tzset function
 #  -DHAVE_UNISTD_H=0 if your compiler lacks a "unistd.h" (Microsoft C++ 7?)
+#  -DMKDIR_TAKES_ONE_ARG if your mkdir takes one arg (MinGW)
 #  -DNO_RUN_TIME_WARNINGS_ABOUT_YEAR_2000_PROBLEMS_THANK_YOU=1
 #	if you do not want run time warnings about formats that may cause
 #	year 2000 grief
diff -urN tz-2016a~orig/private.h tz-2016a/private.h
--- tz-2016a~orig/private.h	2015-09-15 08:36:31.000000000 +0200
+++ tz-2016a/private.h	2016-02-23 10:58:26.931371474 +0100
@@ -58,6 +58,11 @@
 #define HAVE_UTMPX_H		1
 #endif /* !defined HAVE_UTMPX_H */
 
+#ifdef MKDIR_TAKES_ONE_ARG
+#include <sys/stat.h>
+#define mkdir(path,mode) mkdir(path)
+#endif
+
 #ifndef NETBSD_INSPIRED
 # define NETBSD_INSPIRED 1
 #endif
