--- ntp-4.2.6p2/ntpd/ntp_timer.c~	2009-12-09 08:36:35.000000000 +0100
+++ ntp-4.2.6p2/ntpd/ntp_timer.c	2010-12-27 20:33:47.653030402 +0100
@@ -131,15 +131,15 @@
 	if (itimer.it_value.tv_sec < 0 || itimer.it_value.tv_sec > (1<<EVENT_TIMEOUT)) {
 		itimer.it_value.tv_sec = (1<<EVENT_TIMEOUT);
 	}
-	if (itimer.it_value.tv_usec < 0 ) {
-		itimer.it_value.tv_usec = 0;
+	if (itimer.it_value.tv_nsec < 0 ) {
+		itimer.it_value.tv_nsec = 0;
 	}
-	if (itimer.it_value.tv_sec == 0 && itimer.it_value.tv_usec == 0) {
+	if (itimer.it_value.tv_sec == 0 && itimer.it_value.tv_nsec == 0) {
 		itimer.it_value.tv_sec = (1<<EVENT_TIMEOUT);
-		itimer.it_value.tv_usec = 0;
+		itimer.it_value.tv_nsec = 0;
 	}
 	itimer.it_interval.tv_sec = (1<<EVENT_TIMEOUT);
-	itimer.it_interval.tv_usec = 0;
+	itimer.it_interval.tv_nsec = 0;
 	setitimer(ITIMER_REAL, &itimer, (struct itimerval *)0);
 #  endif
 # endif /* VMS */
@@ -191,7 +191,7 @@
 #  else
 	(void) signal_no_reset(SIGALRM, alarming);
 	itimer.it_interval.tv_sec = itimer.it_value.tv_sec = (1<<EVENT_TIMEOUT);
-	itimer.it_interval.tv_usec = itimer.it_value.tv_usec = 0;
+	itimer.it_interval.tv_nsec = itimer.it_value.tv_nsec = 0;
 	setitimer(ITIMER_REAL, &itimer, (struct itimerval *)0);
 #  endif
 # else /* VMS */
