diff -urN gettext-0.18.1.1/build-aux/install-reloc gettext-0.18.1.1.mod/build-aux/install-reloc
--- gettext-0.18.1.1/build-aux/install-reloc	2012-10-08 15:46:56.000000000 +0200
+++ gettext-0.18.1.1.mod/build-aux/install-reloc	2012-10-08 15:47:01.000000000 +0200
@@ -160,6 +160,8 @@
              "$srcdir"/canonicalize-lgpl.c \
              "$srcdir"/malloca.c \
              "$srcdir"/relocatable.c \
+             "$srcdir"/lstat.c \
+             "$srcdir"/stat.c \
              "$srcdir"/setenv.c \
              "$srcdir"/strerror.c \
              "$srcdir"/c-ctype.c \
diff -urN gettext-0.18.1.1/gettext-tools/Makefile.in gettext-0.18.1.1.mod/gettext-tools/Makefile.in
--- gettext-0.18.1.1/gettext-tools/Makefile.in	2010-06-06 22:12:33.000000000 +0200
+++ gettext-0.18.1.1.mod/gettext-tools/Makefile.in	2012-10-08 14:54:25.000000000 +0200
@@ -1248,7 +1248,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.5 gnu no-dependencies
 ACLOCAL_AMFLAGS = -I m4 -I ../gettext-runtime/m4 -I ../m4 -I gnulib-m4 -I libgrep/gnulib-m4 -I libgettextpo/gnulib-m4
-SUBDIRS = doc intl gnulib-lib libgrep src libgettextpo po projects styles misc man m4 tests gnulib-tests examples
+SUBDIRS = doc intl gnulib-lib libgrep src libgettextpo po projects styles misc man m4 tests gnulib-tests
 
 # Allow users to use "gnulib-tool --update".
 
diff -urN gettext-0.18.1.1/gettext-tools/configure gettext-0.18.1.1.mod/gettext-tools/configure
--- gettext-0.18.1.1/gettext-tools/configure	2010-06-06 22:12:20.000000000 +0200
+++ gettext-0.18.1.1.mod/gettext-tools/configure	2012-10-08 14:54:59.000000000 +0200
@@ -40606,7 +40606,9 @@
 #include <stdlib.h>
 #include <string.h> /* for strcpy */
 /* The stpncpy prototype is missing in <string.h> on AIX 4.  */
+#ifndef stpncpy 
 extern char *stpncpy (char *dest, const char *src, size_t n);
+#endif
 int main () {
   const char *src = "Hello";
   char dest[10];
diff -urN gettext-0.18.1.1/gettext-tools/gnulib-lib/stpncpy.c gettext-0.18.1.1.mod/gettext-tools/gnulib-lib/stpncpy.c
--- gettext-0.18.1.1/gettext-tools/gnulib-lib/stpncpy.c	2010-05-24 11:42:47.000000000 +0200
+++ gettext-0.18.1.1.mod/gettext-tools/gnulib-lib/stpncpy.c	2012-10-08 15:00:12.000000000 +0200
@@ -25,7 +25,7 @@
 #include <string.h>
 
 #ifndef weak_alias
-# define __stpncpy stpncpy
+//# define __stpncpy stpncpy
 #endif
 
 /* Copy no more than N bytes of SRC to DST, returning a pointer past the
