--- lttng-modules-2.0-pre11/Makefile.orig	2012-01-18 08:53:35.315220087 +0100
+++ lttng-modules-2.0-pre11/Makefile	2012-01-18 08:53:58.996459403 +0100
@@ -46,7 +46,7 @@
 	$(MAKE) -C $(KERNELDIR) M=$(PWD) modules
 
 modules_install:
-	$(MAKE) -C $(KERNELDIR) M=$(PWD) modules_install
+	$(MAKE) -C $(KERNELDIR) M=$(PWD) INSTALL_MOD_PATH=$(INSTALL_MOD_PATH) modules_install
 
 clean:
 	$(MAKE) -C $(KERNELDIR) M=$(PWD) clean
