--- lttng-tools-2.0-pre16/lttng-sessiond/main.c.orig	2012-01-18 13:46:22.565699529 +0100
+++ lttng-tools-2.0-pre16/lttng-sessiond/main.c	2012-01-18 13:47:27.649169500 +0100
@@ -1528,9 +1528,9 @@
 		 * Exec consumerd.
 		 */
 		if (opt_verbose > 1 || opt_verbose_consumer) {
-			verbosity = "--verbose";
+			verbosity = "-";
 		} else {
-			verbosity = "--quiet";
+			verbosity = "-q";
 		}
 		switch (consumer_data->type) {
 		case LTTNG_CONSUMER_KERNEL:

