diff -urN busybox-1.20.2/scripts/kconfig/mconf.c busybox-1.20.2-2/scripts/kconfig/mconf.c
--- busybox-1.20.2/scripts/kconfig/mconf.c	2012-02-04 20:24:55.000000000 +0100
+++ busybox-1.20.2-2/scripts/kconfig/mconf.c	2012-10-04 09:38:35.000000000 +0200
@@ -9,6 +9,8 @@
  */
 
 #define _XOPEN_SOURCE 700
+/* On Darwin, this may be needed to get SIGWINCH: */
+#define _DARWIN_C_SOURCE 1
 
 #include <sys/ioctl.h>
 #include <sys/wait.h>
