--- util-linux-2.22.2/configure.ac~orig	2013-04-25 13:52:40.527984112 +0200
+++ util-linux-2.22.2/configure.ac	2013-04-25 13:56:19.579715808 +0200
@@ -70,10 +70,14 @@
 
 
 # The original default values of {bin,sbin,lib}dir
-usrbin_execdir='${exec_prefix}/bin'
+if test -z "${usrbin_execdir}" ; then
+  usrbin_execdir='${exec_prefix}/'`basename "$bindir"`
+fi
 AC_SUBST([usrbin_execdir])
 
-usrsbin_execdir='${exec_prefix}/sbin'
+if test -z "${usrsbin_execdir}" ; then
+  usrsbin_execdir='${exec_prefix}/'`basename "$sbindir"`
+fi
 AC_SUBST([usrsbin_execdir])
 
 case $libdir in
